extra_foam/gui/image_tool/reference_view.py

```diff
@@ -76,26 +103,107 @@ def _loadReference(self):
         if filepath:
             self._pub.set(filepath)
             self._ctrl_widget.filepath_le.setText(filepath)
+            self._ctrl_widget.save_btn.setDisabled(True)
+            # Update label
+            text = "The loaded file has been applied as reference."
+            self._ctrl_widget.record_label.setText(text)
 
     @pyqtSlot()
-    def _setReference(self):
+    def _setCurrentAsReference(self):
         """Set the current corrected image as reference."""
-        img = self._corrected.image
-        if img is not None:
-            filepath = osp.join(ROOT_PATH, "tmp", ".reference.npy")
-            if not osp.exists(osp.dirname(filepath)):
-                os.mkdir(osp.dirname(filepath))
-
-            try:
-                write_image(filepath, img)
-            except ValueError as e:
-                logger.error(str(e))
-
-            self._pub.set(filepath)
-            self._ctrl_widget.filepath_le.setText(filepath)
+        image = self._corrected.image
+        if image is not None:
+            self._write_temp_reference_file(image)
+            if self._check_reference_file():
+                self._set_reference(REFERENCE_FILE)
+                # Update label
+                text = "The current image has been applied as reference."
+                self._ctrl_widget.record_label.setText(text)
+
+    def _write_temp_reference_file(self, image):
+        """Save the input image as a reference file (.npy).
+           Path is $EXTRA_FOAM_TEMP_FOLDER/.reference.npy"""
```

Comment:
Nitpicking, this could just point to the `REFERENCE_FILE` variable.

Reply:
Changed, thanks for the suggestion :D